### PR TITLE
Complete docs for abs, min, max, sign, clamp

### LINKED ISSUE #34351: Incomplete abs() docs and editor hints
**Godot version:** v3.2.beta.custom_build.3a9284b56

**OS/device including version:** macOS Catalina 10.15.1 / MacBook Pro (Retina, 13-inch, Early 2015)

**Issue description:**

The argument of `abs()` can be either an `int` or `float`, and the function returns a value of the corresponding type.

```gdscript
var i = abs(1)
var f = abs(1.0)
assert(typeof(i) == TYPE_INT)
assert(typeof(f) == TYPE_REAL)
```

Although this behavior is partially mentioned in the function description, it only shows the `float` version of function signature in the documentation and editor hints:

Screenshot of abs function in docs and editor hints

So there are actually two `abs` functions, but currently it seems complicated to change the documentation. Because the two versions are made different by code under the same `GDScriptFunctions::Function::MATH_ABS` enum, and the signature in documentation is generated per `GDScriptFunctions::Function` enum.

p.s. The same goes for `sign()`, `min()`, `max()`, and `clamp()` too.

### PR DESCRIPTION
**Pull request by lakshay-angrish**
closes #34351

### CODE REVIEW

File: modules/gdscript/gdscript_functions.cpp [hunk 1]
```diff
@@ -267,14 +272,28 @@ void GDScriptFunctions::call(Function p_func, const Variant **p_args, int p_arg_
...
 		} break;
-		case MATH_ABS: {
+		case MATH_ABS_INT: {
 			VALIDATE_ARG_COUNT(1);
 			if (p_args[0]->get_type() == Variant::INT) {
 
 				int64_t i = *p_args[0];
 				r_ret = ABS(i);
+				break;
 			} else if (p_args[0]->get_type() == Variant::REAL) {
```
Comment (bruvzg) [inline modules/gdscript/gdscript_functions.cpp]: Fall-through branches should explicitly use `FALLTHROUGH;` macro (assuming that's what all the empty else branches are intended for).

Comment (lakshay-angrish) [reply modules/gdscript/gdscript_functions.cpp]: I have made the suggested changes in the new commit

**Comment by Calinou** [comment]
There are extraneous commits included. Could you squash this to a single commit containing only the documentation changes? See [PR workflow](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase) for instructions :slightly_smiling_face:

**Comment by lakshay-angrish** [comment]
@Calinou I think I messed up with the commit history on this one. I think it would be better if I close this PR and make a new one.

**Comment by Calinou** [comment]
@lakshay-angrish If you know how to use the Git command line, you could do the following to avoid closing and opening a new PR:

1. Make a copy of your documentation changes.
2. Checkout your branch on Git (`git checkout improve-abs-doc`), go back in the history by the number of commits in your PR (`git reset --hard HEAD~4`).
3. Copy your documentation changes onto the now-pristine branch.
4. Make a commit with your changes.
5. Force-push your changes to your fork (`git push -f`).

You could also open a new pull request, but this is a cleaner approach :slightly_smiling_face:

**Pull request closed without merging**